consensus/types/src/chain_spec.rs

```diff
@@ -712,6 +711,33 @@ impl ChainSpec {
         }
     }
 
+    /// Worst-case compressed length for a given payload of size n when using snappy.
+    ///
+    /// https://github.com/google/snappy/blob/32ded457c0b1fe78ceb8397632c416568d6714a0/snappy.cc#L218C1-L218C47
+    /// https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#max_compressed_len
+    #[allow(clippy::arithmetic_side_effects)]
+    fn max_compressed_len_snappy(n: usize) -> usize {
+        32 + n + (n / 6)
+    }
```

Comment:
shouldn't we add guard rails here to not overflow? 
I.e. 
```suggestion
    fn max_compressed_len_snappy(n: usize) -> Option<usize> {
        32_usize.checked_add(n)?.checked_add(n/6)
    }
```

and then `unwrap_or_()` or `expect` with a message when using it

Reply:
Yeah sorry was being lazy. Made it safe arithematic and added a test to make sure default values don't overflow